mantidproject/mantid
# Data Processor Refactoring Part 1

This PR begins the refactoring the `DataProcessor` classes as described by the issue #20665 .

To summarise this PR
- Adds a `const_iterator` to the `WhiteList` class.
- Groups postprocessing and preprocessing attributes of `GenericDataProcessorPresenter` into their own classes, `PreprocessingAttributes` and `PostprocessingStep`.
- Moves the postprocessGroup routine out of `GenericDataProcessorPresenter` and into `PostprocessingStep`.
- Replaces the postprocessing related member variables with a `optional<PostprocessingStep>` member variable`
- Removes the `m_columns` and `m_postprocess` member variables.

**To test:**
- Ensure that the SANS v2 and Reflectometry GUIs open and continue to operate without errors.
- Ensure the unit tests pass.
- Code Review

No Issue

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
